src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorDocumentManager.ts

```diff
@@ -124,10 +124,11 @@ export class RazorDocumentManager implements IRazorDocumentManager {
 
     private _getDocument(uri: vscode.Uri) {
         const path = getUriPath(uri);
-        const document = this.razorDocuments[path];
+        let document = this.razorDocuments[path];
 
+        // This might happen in the case that a file is opened outside the workspace
         if (!document) {
-            throw new Error('Requested document does not exist.');
+            document = this.addDocument(uri);
```

Comment:
`logMessage` here to indicate that we weren't previously tracking this document and believe it's from outside of the workspace. Will be useful to have a log in place if users end up having issues that we weren't expecting.

Reply:
Will we ever need to remove the document from our tracking? E.g, if someone keep opening and closing a bunch of files from outside the workspace, will be end up tracking all of those?

Reply:
Yes, if someone opens and closes a ton of documents outside of the workspace (say, all of them) then we would continue to track those files until the instance is closed. If we have evidence that that's going to be a perf concern we could start removing out of workspace files from the tracked list when they're closed, but my instinct is that opening any files outside the workspace is a niche scenario, so we're mostly concerned with it just not throwing in this case.

@NTaylorMullen thoughts? Can you think of any dangers to "forgetting" files outside the workspace that are no longer opened?

Reply:
I can't foresee any issues with removing files that are out of the workspace when they're closed. 

Let's do our best not to leak memory even if it's a niche like scenario.

Reply:
I agree. We should not track closed non-workspace files.